index.js

```diff
@@ -224,21 +225,40 @@ async function terraform(terraformDirectory, args) {
       exitCode = 1;
       break;
     }
+    core.setOutput('changes', tfd.status == 2 ? 'true' : 'false');
     core.endGroup();
     process.exit(exitCode);
   }
 
   core.startGroup('Run terraform plan');
-  const tfp = await terraform(terraformDirectory, ['plan', `-lock=${terraformLock}`, `-parallelism=${terraformParallelism}`, `-refresh=${terraformRefresh}`, '-out=terraform.plan'].concat(terraformTargets).concat(terraformVariableFiles).concat(terraformPlanDestroy));
+  const tfp = await terraform(terraformDirectory, ['plan', `-lock=${terraformLock}`, `-parallelism=${terraformParallelism}`, `-refresh=${terraformRefresh}`, '-out=terraform.plan', '-detailed-exitcode'].concat(terraformTargets).concat(terraformVariableFiles).concat(terraformPlanDestroy));
   core.endGroup();
-  if (tfp.status > 0) {
+  if (tfp.status == 1) { // 0 = no error no changes, 1 = error, 2 = no error with changes (from -detailed-exitcode documentation)
     tf_plan = status_failed;
     core.setFailed(`Failed to prepare the terraform plan [err:${tfp.status}]`);
     terraformDoApply = false;
   } else {
     tf_plan = status_success;
   }
 
+  core.setOutput('changes', tfp.status == 2 ? 'true' : 'false');
+
+  // write to file and upload to artifact storage
+  const fileName = terraformDirectory.replace('./', '').slice(0, -1).concat(terraformBackend ? '.' + path.parse(terraformBackend).name + '.summary' : '.summary');
```

Comment:
The `slice()` assumes `terraformDirectory` to end with a `/`. Then again, it would only cause actual issues if only the last char would be unique in the path :shrug: Still feels a bit icky though. But the alternatives I could find (explicitly stripping trailing slashes) are way more cumbersome :disappointed: